tests/SoldiersAPI.test.ts

```diff
@@ -0,0 +1,42 @@
+import request from 'supertest';
+import router from '../src/routes/routes';
+import * as soldiers from './testData';
+import { getDatabase, connectToMongoDB } from '../src/repository/mongoConnect';
+
+const app = router;
+beforeEach(async () => {
+  await connectToMongoDB();
+  const db = await getDatabase();
+  await db.collection('soldiers').deleteMany({});
+});
+describe('Insert Soldier into DB', () => {
+  it('Should Send an API Post Request and save soldier to DB, Expected 201:', async () => {
+    await request(app).post('/soldiers').send(soldiers.SoldierToBeAddedForPOSTtest).expect(201);
+  });
+});
+
+describe('Fail to insert Soldier into DB', () => {
+  it('Should Send an API Post Request and not save soldier to DB, Expected 400:', async () => {
+    await request(app).post('/soldiers').send(soldiers.SoldierToBeAddedForFailedPOSTtest).expect(400);
+  });
+});
+
+describe('Find Soldier By ID', () => {
+  it('Should Send an API Get Request If Soldier Is In DB, Expected 200:', async () => {
+    request(app).post('/soldiers/fuck').send(soldiers.SoldierToBeFound);
```

Comment:
when you want to insert a soldier when testing get - just insert it directly into mongo in a beforeEach.

This is because you are testing the insert as well as the find here - what happens if there is a problem with the `insert` function?